ddtrace/propagation/http.py

```diff
@@ -197,11 +210,18 @@ def _inject(span_context, headers):
     @staticmethod
     def _extract(headers):
         # type: (Dict[str, str]) -> Optional[Context]
-        trace_id = _extract_header_value(
-            POSSIBLE_HTTP_HEADER_TRACE_IDS,
-            headers,
-        )
-        if trace_id is None:
+        trace_id_str = _extract_header_value(POSSIBLE_HTTP_HEADER_TRACE_IDS, headers)
+        if trace_id_str is None:
+            return None
+        try:
+            trace_id = int(trace_id_str)
+        except ValueError:
+            trace_id = 0
+
+        if trace_id == 0 or trace_id > _MAX_UINT_64BITS:
+            log.debug(
```

Comment:
```suggestion
            log.warning(
```

probably want to warn here since this would be broken trace propagation